piptools/scripts/sync.py

```diff
@@ -9,15 +10,21 @@
 from pip._internal.commands.install import InstallCommand
 from pip._internal.index.package_finder import PackageFinder
 from pip._internal.utils.misc import get_installed_distributions
+from pip._vendor.packaging.version import Version
```

Comment:
Have you checked if `packaging` is available directly, maybe as a transitive dep?

Reply:
It is available, but because other parts of code are using imports from `pip._vendor`, mixing with direct imports will break `isinstance` like checks. Not sure about the original motivation to use  `pip._vendor` imports, most probably it is to make it more stable and not rely on libraries except pip